Sources/Safetensors/Safetensors.swift

```diff
@@ -208,21 +17,32 @@ public enum Safetensors {
     static func validate(header: [String: HeaderElement], dataCount: Int) throws {
         let allDataOffsets = header
             .values
-            .compactMap { $0.tensorData?.dataOffsets }
-            .sorted { $0.start < $1.start }
+            .compactMap {
+                $0.tensorData?.dataOffsets
+            }
+            .sorted {
+                $0.start < $1.start
+            }
+        
         guard let first = allDataOffsets.first, let last = allDataOffsets.last else {
             throw SafetensorsError.metadataIncompleteBuffer
         }
+        
         if first.start != 0 || last.end != dataCount {
             throw SafetensorsError.metadataIncompleteBuffer
         }
+        
+        if zip(allDataOffsets, allDataOffsets.dropFirst()).contains(where: { $0.end != $1.start }) {
+            throw SafetensorsError.metadataIncompleteBuffer
+        }
+        
         for (first, second) in zip(allDataOffsets, allDataOffsets.dropFirst()) {
             if first.end != second.start {
                 throw SafetensorsError.metadataIncompleteBuffer
             }
         }
```

Comment:
this probably repeats what the for loop is already doing, I think for loop is clearer here